src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,38 +1,83 @@
+import { useEffect, useState } from 'react';
+
+//components
+
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  selectedTodo: Todo;
+  setSelectedTodo: () => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  selectedTodo,
+  setSelectedTodo,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [userData, setUserData] = useState<User>(null);
+  const [hasErrorMessage, setErrorMessage] = useState('');
+
+  const userId = selectedTodo.userId;
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    if (userId) {
+      getUser(userId)
+        .then(data => setUserData(data))
+        .catch(e => setErrorMessage(e))
```

Comment:
Consider providing a more descriptive error message in the `catch` block to help with debugging. For example, `setErrorMessage('Failed to load user data: ' + e.message)`.